skimage/feature/texture.py

```diff
@@ -97,11 +99,21 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
```

Comment:
Thanks for all the comments. 

I think it is actually better to have `levels` adaptive. The function generates a `levels x levels` output matrix (for each distance and each angle). If the maximum value << `levels`, there will many zeros. An alternative would be to use sparse matrices. In any case, as a user I would not expect that the co-occurrence matrix covers the entire range defined by the data type unless I wish to combine the co-occurrence matrices of many images, which seems a bit exotic as an application (and which is still feasible, even if `levels` is adaptive. 

But if there is a strong incentive to have the matrices cover the entire data range, I can of course change this accordingly. Using the max just seems more logical to me.